### ISSUE TITLE
Saved filters (& desktop shortcuts & back button) should respect selected todo.txt file

### ISSUE THREAD

**Issue by tomtom** [mpcjanssen/simpletask-android#293]
When I

1. Open a todo.txt file A
2. Define a saved filter X
3. Open another todo.txt file B
4. Select the saved filter

then the original todo.txt A file isn't restored but the filter X is applied to file B. IMHO simpletask would benefit from a different behaviour: Selecting filter X should always open file A. Selecting the respective shortcut from the "desktop" should also open file A. Pressing the back button shouldn't just remove the filter but also open the previous file if applicable.

**Comment by mpcjanssen** [comment]
The todo file name is not part of the filter. Without a big redesign I don't think it's easy to implement this. Also:

1) If I add todo filename to the filter, how will you use the same filter for different files? Would you need to create different saved filters? That sounds like a lot of work.
2) If filters have a filename how would this work for widgets?

I am planning to make a recent files list below the save filters. This would work for quickly switching as well.

Can I also suggest to combine everything in a single file and split the different parts with a new `@list` or a new `+tag` ?

Or in other words, why do you use different files instead of different lists?

**Comment by tomtom** [comment]
> why do you use different files instead of different lists?

In one file, I want to archive obsolete tasks. In the other file, I want to keep obsolete/done tasks. Let's say the "one file" is a todo list and the "other file" is, e.g., a shopping list.

Why did you implement support for using multiple todo files and even linking/jumping between those files?

ad "1 same filter for different files":
IMHO this is an irrelevant use case. I want to use different files (not just different lists) because the nature of the lists is fundamentally different. I thus also use different tags & keywords in those files. Thus there hardly are any filters I would want to use in both files.

ad 2:
I hardly use the widget. The file should be defined when defining the widget. Opening another file in simpletask shouldn't IMHO have any affect on the widget that was defined months earlier.

**Comment by mpcjanssen** [comment]
> In one file, I want to archive obsolete tasks. In the other file, I want to keep obsolete/done tasks. 
> Let's say the "one file" is a todo list and the "other file" is, e.g., a shopping list.

Makes sense. I can also imagine checklists where you don't want to archive completed tasks.

> Why did you implement support for using multiple todo files and even linking/jumping between those files?

Some people like to switch to project specific todo files easily. The opening another file was never intended for switching. It initially was only used to open the todo file if was not called `todo.txt`.

> ad "1 same filter for different files":
> IMHO this is an irrelevant use case. I want to use different files (not just different lists) because 
> the nature of the lists is fundamentally different. I thus also use different tags & keywords in those > files. Thus there hardly are any filters I would want to use in both files.

For me this is a very common use case. Problem with a format as flexible as todo.txt is that there are a lot of different use cases. Any change pretty much breaks someones usecase.

> ad 2:
> I hardly use the widget. The file should be defined when defining the widget. Opening another file > in simpletask shouldn't IMHO have any affect on the widget that was defined months earlier.

I agree completely, however this is not as simple as it sounds, because the widgets should update when the todo files update. When the widget contains the filename also, Simpletask needs to monitor a lot of files in the background and this gets tricky quickly.

I really don't know a good way to support multiple todo files really elegantly yet. I am open to ideas though.

**Comment by tomtom** [comment]
Alternative proposal/RFE: What about adding a widget that opens simpletask and opens a file (similar to the "shortcut" widget that opens simpletask and applies a filter). It would make working with multiple files sufficiently easy.

**Comment by mpcjanssen** [comment]
I like this proposal much better.  Shortcut handling wel have an optional filename.  This can be applied both to shortcuts and saved filters. So when defining the saved filter you'll be asked if the filename should be included. And the shortcut just keeps the setting from the saved filter.

**Comment by mpcjanssen** [comment]
I will add this to Simpletask as soon as I have time. It should be a fairly easy change.

**Comment by tomtom** [comment]
Thank you very much!

**Comment by flight16** [comment]
That solution sounds great.  I reported the same thing in #282 .  I can't wait to see this implemented.

**Comment by y2kbugger** [comment]
In addition to allowing a filter to be associated with a specific file, it should then have the option to be shown globally (any file open), or locally (only when the associated files is open) that way you can use filters for jumping between files, and also have filters that can "dig deeper" into certain files without cluttering the global list.

**Comment by bshanks** [comment]
+1
